core/modules/qana/QservRestrictorPlugin.cc

```diff
@@ -571,14 +574,36 @@ QservRestrictorPlugin::_newSecIndexRestrictor(
         return query::QsRestrictor::Ptr();
     }
 
+    // Get the director table (or an empty string)
+    std::string dirTable = context.cssFacade->getDirTable(cr->db, cr->table);
+
     // Build the QsRestrictor
     query::QsRestrictor::Ptr p = std::make_shared<query::QsRestrictor>();
     p->_name = "sIndex";
     // sIndex has paramers as follows:
     // db, table, column, val1, val2, ...
-    p->_params.push_back(cr->db);
-    p->_params.push_back(cr->table);
-    p->_params.push_back(cr->column);
+    if (!dirTable.empty()) {
+        std::string dirDb = context.cssFacade->getDirDb(cr->db, cr->table);
+        if (dirDb.empty()) {
+            dirDb = cr->db;
+        }
+        std::string dirCol = context.cssFacade->getDirColName(cr->db, cr->table);
+        if (dirCol.empty()) {
+            LOGF_ERROR("dirTable set, dirCol missing for %1%.%2%" % cr->db % cr->table);
+            return query::QsRestrictor::Ptr();
+        }
+        LOGF_DEBUG("Using dirDb %1%, dirTable %2%, dirCol %3% as sIndex for %4%.%5%." %
+                   dirDb % dirTable % dirCol % cr->db % cr->table);
+        p->_params.push_back(dirDb);
+        p->_params.push_back(dirTable);
+        p->_params.push_back(dirCol);
+    } else {
+        LOGF_DEBUG("Using %1%, %2%, %3% as sIndex." %
+                   cr->db % cr->table % cr->column);
+        p->_params.push_back(cr->db);
+        p->_params.push_back(cr->table);
+        p->_params.push_back(cr->column);
+    }
```

Comment:
Serge we are both wrong. We have both secIndexColName and dirColName per table. We really need to clean that up. How about the following:
 * get rid of secIndexColName
 * if table is a director table, dirColName must be set, and dirColTable must not be set and dirDb must not be set
 * if the table is not a director table, dirTable must be set, optionally dirDb maybe set, dirColName must not be set.